OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseDetailFragment.java

```diff
@@ -303,6 +303,10 @@ private void configureEnrollButton() {
 
         if (mEnrolled) {
             mEnrollButton.setText(R.string.view_course_button_text);
+        } else if (courseDetail.invitation_only) {
```

Comment:
I second miankhalid here, as `invitation_only` has `Boolean` data type, if it will not come in server call it will throw NPE on default `null` value, must have a null check here.